### PR TITLE
Development add long lat to location

### PR DESCRIPTION
**Pull request by sameh-farouk**
### Description

- Adding Longitude and Latitude to Node Location struct.
- Update swagger doc.

### Changes

- Modifying the  SQL node query.
- Adding **longitude** and **latitude** to node location struct.
- Adding custom SQL function to cast the text type to decimal and return null in case of empty string/invalid entry.
- Both values Return from API as JSON **Numbers** and could be **Null**. **clients who consume the API should handle the null value.**

### Related Issues

https://github.com/threefoldtech/tfgridclient_proxy/issues/132
https://github.com/threefoldtech/tfgrid_dashboard/issues/410

### Checklist

- [ ] Tests included
- [X] Build pass
- [X] Documentation
- [X] Code format and docstrings

### CODE REVIEW

File: internal/explorer/db/types.go [hunk 1]
```diff
@@ -59,6 +61,8 @@ type Node struct {
 	RentContractID  int64
 	RentedByTwinID  int64
 	SerialNumber    string
+	Longitude       *float64
```
Comment (xmonader) [inline internal/explorer/db/types.go:64]: I don't believe they should be pointers, and in the database that longitude and latitude columns shouldn't allow nulls 

I checked in the graphql https://graphql.grid.tf/graphql


```
query MyQuery {
  countries {
    id
    lat
    long
  }
}

```
Don't believe they should be nullables

Comment (xmonader) [reply internal/explorer/db/types.go:64]: I also checked in the database scheme 
image

seems the columns aren't allowing null as well.

Comment (sameh-farouk) [reply internal/explorer/db/types.go:64]: @xmonader Thanks for your review and information.
Let me clarify my decisions.

1. As you observed, long and lat are text types in the DB, and it looked ugly to me to return these supposedly decimal data from the API as it is(text), it would be required further conversion from the clients. So I decide to return it as JSON numbers, and for this, I needed to convert the text to decimal.

3. I observed, by looking at the DB data, that there is some missing/invalid long, and lat data (not null, but empty text), that needs to be handled properly when casting to numbers, can't use an empty numeric value, 0, so I returned on purpose null in these cases as a more suitable choice.

4. The reason that I don't want to use an empty value for the numeric type, 0, to represent the missing/invalid entry is that 0 would consider a valid long, and lat.

5. This leads me to the decision of using *float64, to keep nullable values (after converting from text) from converted to empty float64 value 0 as struct gets filled.

To sum it up,
Originally long, lat data are text types with some entries having empty text (missing/invalid)
I decided to make API return the data as number types with some entries having null to represent missing/invalid values on original data.

**Review by xmonader — APPROVED** [review]
